app/src/features/Mappings/mappingSlice.ts

```diff
@@ -0,0 +1,102 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { v4 as uuid } from "uuid";
+
+import { RootState } from "app/store";
+import { Column, Filter, Resource } from "services/api/generated/api.generated";
+
+export type PendingJoin = {
+  id: string;
+  columns: [Partial<Column>, Partial<Column>];
+};
+
+export type PendingFilter = Partial<Filter & { col: Partial<Column> }>;
+
+type MappingState = {
+  current?: Partial<Resource>;
+  filters?: PendingFilter[];
+  joins?: Record<string, PendingJoin[]>;
+};
+
+const initialState: MappingState = {};
+
+const mappingSlice = createSlice({
```

Comment:
since we have mostly crud actions here. how about using this: https://redux-toolkit.js.org/api/createEntityAdapter ?